src/RpcClient.ts

```diff
@@ -354,4 +378,94 @@ export class RpcClient {
       throw new Error(`Error in getSignaturesForAsset: ${error}`);
     }
   }
+  /**
+   * Mint a compressed NFT.
+   * @returns {Promise<MintCNFTResponse>}
+   * @throws {Error}
+   */
+  async mintCNFT(
+    request: MintCNFTRequest
+  ): Promise<MintCNFTResponse | undefined> {
+    try {
+      let ownerKeypair = loadWallet(request.ownerKeypair);
+      let defaultMetadataArgs: MetadataArgs = {
+        ...request.metadataArgs,
+        primarySaleHappened: false,
+        isMutable: false,
+        editionNonce: 0,
+        tokenStandard: null,
+        uses: null,
+        tokenProgramVersion: null,
+      };
+      let treeWallet: Keypair;
+      if (!request.treeKeypair) {
```

Comment:
This isn't optimal UX. If they pass in a Merkle tree then you should use the one they create.

If you want to create a seamless approach, then use a PDA to derive the Merkle tree based on the owner keypair and then you can consistently mint to the same tree.

Reply:
Why did we opt to not go the PDA route?

If you create a tree like this you'll end up having a dead tree, because the user will never be able to access it.

Reply:
My thinking here was to include the merkle tree keypair for the response, since the PDA route was typing the keypair to just one merkle tree where they might want to be more flexible.